somatic_llr_filter.py

```diff
@@ -0,0 +1,302 @@
+        "--sequence-error-rate",
+        help="expected sequencing error rate (range 0 to 1) - default 0.005",
+        type=float,
+        default=0.005
+    )
+    parser.add_argument(
+        "--tumor-purity",
+        help="tumor purity (range 0 to 1) - default 1",
+        type=float,
+        default=1
+    )
+    parser.add_argument(
+        "--normal-contamination-rate",
+        help="normal contamination rate (range 0 to 1) - default 0",
+        type=float,
+        default=0
+    )
+    parser.add_argument(
+        "--allele-depth-field",
+        help="field corresponding to allele depth - default AD",
+        default="AD"
+    )
+    parser.add_argument(
+        "--site-depth-field",
+        help="field corresponding to site depth - default DP",
+        default="DP"
+    )
+    parser.add_argument(
+        "--tumor-sample-name",
+        help="name of the vcf sample corresponding to the tumor - default TUMOR",
+        default="TUMOR"
+    )
+    parser.add_argument(
+        "--normal-sample-name",
+        help="name of the vcf sample corresponding to the normal - default NORMAL",
+        default="NORMAL"
+    )
+    parser.add_argument(
+        "--llr-field",
+        help="name of the vcf info field in which to store the log-likelihood ratio value",
+        default="LLR"
+    )
+    parser.add_argument(
+        "--somatic-field",
+        help="name of the vcf info field in which to store the classification call from this script (somatic = 0 or 1)",
+        default="SOMATIC"
+    )
+    parser.add_argument('-w', "--overwrite", action='store_true',
+        help="by default, this tool will raise an exception if the LLR or SOMATIC fields already exist in the VCF. This flag allows existing fields to be overwritten."
+    )
+    return parser
+
+
+def create_vcf_reader(args):
+    vcf_reader = vcfpy.Reader.from_path(args.input_vcf)
+    #do some sanity checking
+    sample_names = vcf_reader.header.samples.names
+    is_multi_sample = len(sample_names) > 1
+    if(not is_multi_sample):
+        vcf_reader.close()
+        raise Exception("A multisample VCF with both tumor and normal data is required")
+
+    if not args.tumor_sample_name in sample_names:
+        raise Exception("Could not find tumor sample name {} in sample names".format(args.tumor_sample_name))
+    if not args.normal_sample_name in sample_names:
+        raise Exception("Could not find normal sample name {} in sample names".format(args.normal_sample_name))
+
+    #check for needed format fields
+    if not args.allele_depth_field in vcf_reader.header.format_ids():
+        vcf_reader.close()
+        raise Exception("No " + args.allele_depth_field + " format field found. Annotate your VCF with readcounts first")
+    if not args.site_depth_field in vcf_reader.header.format_ids():
+        vcf_reader.close()
+        raise Exception("No " + args.site_depth_field + " format field found. Annotate your VCF with readcounts first")
```

Comment:
The other Exceptions look like this:
```suggestion
        raise Exception("No {} format field found. Annotate your VCF with readcounts first".format(args.site_depth_field))
```